src/legacy/Logary.Services.Rutta/Ingestion.TCP.fs

```diff
@@ -0,0 +1,49 @@
+module Logary.Ingestion.TCP
+
+
+open System.Text
+open System.IO
+open System.Net
+open Logary
+open Logary.Message
+open Logary.Formatting
+open Logary.Internals
+open Logary.Internals.Chiron
+open Hopac
+open fszmq
+
+type TCPConfig =
+  { endpoint: IPEndPoint }
+  static member create endpoint =
+    { endpoint = endpoint }
+
+let private ilogger = Log.create "Logary.Ingestion.TCP"
+
+let rec streamRecvLoop (receiver: Socket) (next: Ingest) =
+  // https://gist.github.com/lancecarlson/fb0cfd0354005098d579
+  // https://gist.github.com/claws/7231548#file-czmq-stream-server-c-L21
+  try
+    let frame = Socket.recv receiver
+    // Aborted socket due to closing process:
+    if isNull frame then () else
+    let bs = Socket.recv receiver
+    // http://api.zeromq.org/4-1:zmq-socket#toc19
+    // A connection was made
+    if bs.Length = 0 then streamRecvLoop receiver next else
+    // printfn "Data received: %A" message
+    use ms = new MemoryStream(bs)
+    use sr = new StreamReader(ms, Encoding.UTF8)
+    while not sr.EndOfStream do
```

Comment:
Slightly broken, see https://stackoverflow.com/questions/524818/what-is-a-good-method-to-handle-line-based-network-i-o-streams/524954#524954 — which is also broken because it assumes ASCII (which it works for, but for UTF8 we may receive partial lines depending on the package sizes)